lib/spack/spack/main.py

```diff
@@ -669,12 +669,7 @@ def main(argv=None):
         cmd_name = args.command[0]
         cmd_name = aliases.get(cmd_name, cmd_name)
 
-        try:
-            command = parser.add_command(cmd_name)
-        except ImportError:
-            if spack.config.get('config:debug'):
-                raise
-            tty.die("Unknown command: %s" % args.command[0])
+        command = parser.add_command(cmd_name)
```

Comment:
I believe so:
```console
[greenc@cluck] ~/work/cet-is/sources/spack-test (feature/enhanced-commands) $ spack ethel
==> Error: ethel is not a recognized Spack command or extension command.
Known commands: activate add arch blame bootstrap build build-env buildcache cd checksum clean clone commands compiler compilers concretize config configure create deactivate debug dependencies dependents diy docs edit env extensions fetch find flake8 gpg graph help info install license list load location log-parse maintainers mirror module patch pkg providers pydoc python reindex release-jobs remove repo resource restage setup spec stage test uninstall unload unuse upload-s3 url use versions view
```

Reply:
The first part of the error message is definitely an improvement, but the second one doesn't look very nice:
![Screenshot from 2019-08-07 16-35-30](https://user-images.githubusercontent.com/4199709/62631727-76d94f80-b931-11e9-97de-7706380f1216.png)

as it may span multiple lines on the screen and is not formatted. What do you think of the simpler:
```console
[greenc@cluck] ~/work/cet-is/sources/spack-test (feature/enhanced-commands) $ spack ethel
==> Error: ethel is not a recognized Spack command or extension command.
You can get a list of valid commands with:

    $ spack help
    $ spack help --all

```
?